src/gl/index_buffer.js

```diff
@@ -29,30 +31,34 @@ class IndexBuffer {
         // The bound index buffer is part of vertex array object state. We don't want to
         // modify whatever VAO happens to be currently bound, so make sure the default
         // vertex array provided by the context is bound instead.
-        if (this.gl.extVertexArrayObject === undefined) {
-            (this.gl: any).extVertexArrayObject = this.gl.getExtension("OES_vertex_array_object");
-        }
-        if (this.gl.extVertexArrayObject) {
-            (this.gl: any).extVertexArrayObject.bindVertexArrayOES(null);
+        if (this.context.extVertexArrayObject) {
+            this.context.bindVertexArrayOES.set(null);
         }
     }
 
     bind() {
-        this.gl.bindBuffer(this.gl.ELEMENT_ARRAY_BUFFER, this.buffer);
+        const gl = this.context.gl;
+        gl.bindBuffer(gl.ELEMENT_ARRAY_BUFFER, this.buffer);
+        // this.context.bindElementBuffer.set(this.buffer);
+        // TODO not sure why this doesn't work, but using bindElementBuffer throws:
```

Comment:
My hunch would be that a necessary call to `gl.bindBuffer()` is getting incorrectly deduped away somehow.